assignments/prototypes.js

```diff
@@ -16,13 +16,87 @@
   * destroy() // prototype method that returns: `${this.name} was removed from the game.`
 */
 
+function GameObject(attr) {
+  this.createdAt = attr.createdAt;
+  this.name = attr.name;
+  this.dimensions = attr.dimensions;
+}
+
+GameObject.prototype.destroy = function() {
+  return `${this.name} was removed from the game.`;
+}
+
 /*
   === CharacterStats ===
   * healthPoints
   * takeDamage() // prototype method -> returns the string '<object name> took damage.'
   * should inherit destroy() from GameObject's prototype
 */
 
+function CharacterStats(attr) {
+  GameObject.call(this, attr);
+  /*
+    *********************** NOTES FOR SELF ******************************
+
+    Output of the above would be something like: {
+      this.createdAt = attr.createdAt;
+      this.name = attr.name;
+      this.dimensions = attr.dimensions;
+    }
+    
+    ...with 'this' being set to the value of the CharacterStats object.
+    
+    Written out this way, you can see more clearly that the properties that 
+    once belonged to GameObject now belong to the CharacterStats object, and 
+    the property values passed to 'attr' now have a place to be extracted and 
+    assigned to.
+
+    *********************** NOTES FOR SELF ******************************
+  */
+
+  this.healthPoints = attr.healthPoints;
+  /* 
+     *********************** NOTES FOR SELF ******************************
+
+     The property above is the only property at this time that explicitly 
+     belongs to the CharacterStats object, although this property can
+     inherited by another just as the GameObject properties were inherited
+
+     *********************** NOTES FOR SELF ******************************
+  */
+}
+
+/** Note: a prototype can have both methods AND properties declared on it **/
+
+CharacterStats.prototype = Object.create(GameObject.prototype);
+
+CharacterStats.prototype.takeDamage = function () {
+  return `${this.name} took damage`;
+}
+
+/*
+  *********************** NOTES FOR SELF ******************************
+
+  The order of the ABOVE two lines of code matters:
+
+  The value of CharacterStats.prototype has to first be set 
+  a copy of the GameObject.prototype object using Object.create().
+
+  If the order is switched, then the method declared on
+  CharacterStats.prototype will be immediately overwritten
+  by the GamObject.prototype object.
```

Comment:
Another student had this same exact problem. We found out about the above together.